File: booksamples/chapter-03/src/main/java/org/vulkanb/eng/graph/vk/VulkanUtils.java
package org.vulkanb.eng.graph.vk;

import java.util.Locale;

import static org.lwjgl.vulkan.VK11.VK_SUCCESS;

public class VulkanUtils {

    private VulkanUtils() {
        // Utility class
    }

    public static OSType getOS() {
        OSType result;
        String os = System.getProperty("os.name", "generic").toLowerCase(Locale.ENGLISH);
        if ((os.indexOf("mac") >= 0) || (os.indexOf("darwin") >= 0)) {
            result = OSType.MACOS;
        } else if (os.indexOf("win") >= 0) {
            result = OSType.WINDOWS;
        } else if (os.indexOf("nux") >= 0) {
            result = OSType.LINUX;
        } else {
            result = OSType.OTHER;
        }

        return result;
    }

    public static void vkCheck(int err, String errMsg) {
        if (err != VK_SUCCESS) {
            throw new RuntimeException(errMsg + ": " + err);
        }
    }

    public enum OSType {WINDOWS, MACOS, LINUX, OTHER}
}

File: booksamples/chapter-05/kotlin/src/main/kotlin/org/vulkanb/eng/graph/vk/VulkanUtils.kt
package org.vulkanb.eng.graph.vk

import org.lwjgl.vulkan.VK11
import org.lwjgl.vulkan.VK11.*

object VulkanUtils {
    val oS: OSType
        get() {
            val result: OSType
            val os = System.getProperty("os.name", "generic").lowercase()
            result = if (os.indexOf("mac") >= 0 || os.indexOf("darwin") >= 0) {
                OSType.MACOS
            } else if (os.indexOf("win") >= 0) {
                OSType.WINDOWS
            } else if (os.indexOf("nux") >= 0) {
                OSType.LINUX
            } else {
                OSType.OTHER
            }
            return result
        }

    fun Int.vkCheck(errMsg: String) {
        if (this != VK_SUCCESS) {
            throw RuntimeException("$errMsg: $this")
        }
    }

    enum class OSType {
        WINDOWS,
        MACOS,
        LINUX,
        OTHER
    }
}
